doc/how_to_guides/benchmarking/benchmarking_tutorial.rst

```diff
@@ -0,0 +1,66 @@
+
+Getting Started
+---------------
+If you haven't already done so, make sure you've completed the steps in :doc:`Getting Started </doc/tutorials/getting_started/getting_started>`.
+
+The :moveit_codedir:`benchmarking package <moveit_ros/benchmarks>` provides methods to benchmark a MoveIt planning pipeline and aggregate/plot statistics using the OMPL Planner Arena.
+The example below demonstrates how the benchmarking can be run.
+
+Example
+-------
+
+To run the example you need to install git lfs by running ``git lfs install`` and clone [moveit_benchmark_resources](https://github.com/ros-planning/moveit_benchmark_resources.git) into your workspace.
+
+Start the benchmarks by running: ::
+
+    ros2 launch moveit2_tutorials run_benchmarks.launch.py
+
+
+This will take a while depending on the settings in ``benchmarks.yaml``. The benchmark results will be saved in ``/tmp/moveit_benchmarks/``.
+To introspect the benchmark data, the log files need to be converted into a database. This can be done using a script provided in the moveit_ros_benchmarks package: ::
+
+    ros2 run moveit_ros_benchmarks moveit_benchmark_statistics.py LOG_FILE_1 ... LOG_FILE_N
+
+This command will create a database containing the data form all benchmark log files included. An easier way to create the database is to create it with all log files from a given repository.
+For example the argument ``/tmp/moveit_benchmarks/*`` can be used to collect all log files in the given directory into a single database. This database is created in the location where the command
+above is run in a file names ``benchmark.db``.
+
+The database can be visualized by uploading the the file to `plannerarena.org <http://plannerarena.org>`_ and interactively visualize the results.
+
+
+.. image:: planners_benchmark.png
+   :width: 700px
+
+ROS 2 parameters to configure a benchmark
+-----------------------------------------
+
+The benchmarks are configured by a set of ROS 2 parameters. You can learn more about these parameters `here <https://github.com/ros-planning/moveit2/blob/main/moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkOptions.h#L50>`
+
+
+The BenchmarkExecutor Class
+---------------------------
+
+This class creates a set of ``MotionPlanRequests`` that respect the parameters given in the supplied instance of ``BenchmarkOptions`` and then executes the requests on each of the planners specified.  From the ``BenchmarkOptions``, queries, ``goal_constraints``, and ``trajectory_constraints`` are treated as separate queries.  If a set of ``start_states`` is specified, each query, ``goal_constraint``, and ``trajectory_constraint`` is attempted with each start state (existing start states from a query are ignored).  Similarly, the (optional) set of path constraints is combined combinatorially with the start query and start ``goal_constraint`` pairs (existing ``path_constraint`` from a query are ignored).  The workspace, if specified, overrides any existing workspace parameters.
```

Comment:
Can this paragraph be split up, turned into a list, or something? It is very dense.